.github/workflows/test.yml

```
@@ -83,3 +83,11 @@ jobs:
           - name: debug
           - name: release
             flag: --release
+
+  nix:
+    runs-on: [ self-hosted, linux ]
+    steps:
+    - uses: actions/checkout@v2.4.0
+    - uses: cachix/install-nix-action@v16
+    - run: nix flake check
+    - run: nix develop -c cargo test --no-default-features --features=backend-kvm --features=wasmldr
```

Maybe consider `--ignore-environment` for a less impure shell environment:

```suggestion
    - run: nix develop --ignore-environment -c cargo test --no-default-features --features=backend-kvm --features=wasmldr
```